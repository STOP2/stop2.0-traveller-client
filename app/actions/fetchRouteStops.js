import config from '../config/config'

export const REQUEST_ROUTE_STOPS = 'REQUEST_ROUTE_STOPS'
export const RECEIVE_ROUTE_STOPS = 'RECEIVE_ROUTE_STOPS'
export const REQUEST_ROUTE_STOPS_ERROR = 'REQUEST_ROUTE_STOPS_ERROR'

export let requestRouteStops = function(tripId)
{
    return {
        type: REQUEST_ROUTE_STOPS,
        isFetching: true,
        tripId: tripId
    }
}

export let receiveRouteStops = function(json)
{
    return {
        type: RECEIVE_ROUTE_STOPS,
        isFetching: false,
        routeIsReady: true,
        error: false,
        stops: json.stops
    }
}

export let requestError = function(error)
{
    return {
        type: REQUEST_ROUTE_STOPS_ERROR,
        sentStoprequest: false,
        error: error
    }
}

export let fetchRouteStops = function(tripId)
{
    return dispatch =>
    {
        dispatch(requestRouteStops(tripId))

        return fetch(config.API_URL + '/routes?trip_id=' + tripId,
        {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        })
      .then(response => response.json())
      .then(json => dispatch(receiveRouteStops(json)))
      .catch(error => dispatch(requestError(error)))
    }
}
