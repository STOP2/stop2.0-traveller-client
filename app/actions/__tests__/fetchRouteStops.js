import mockStore from 'redux-mock-store'

import { fetchRouteStops } from '../fetchRouteStops'

const store = mockStore()

beforeEach(() =>
{
    store.clearActions()
})

describe('fetchRouteStops actions', () =>
{
    it('should handle REQUEST_ROUTE_STOPS action', async() =>
    {
        const response = '{"stops": [{"stop": {"arrives_in": 4, "stop_code": 3028, "stop_id":"HSL:1240118", "stop_name":"Kumpulan kampus"}}]}'

        fetch.mockResponseSuccess(response)

        await store.dispatch(fetchRouteStops('HSL:1240118', '1234', false))

        expect(store.getActions()).toMatchSnapshot()
    })
})
