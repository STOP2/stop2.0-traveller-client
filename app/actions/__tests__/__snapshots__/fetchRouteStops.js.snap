exports[`fetchRouteStops actions should handle REQUEST_ROUTE_STOPS action 1`] = `
Array [
  Object {
    "current": false,
    "isFetching": true,
    "stopId": "1234",
    "tripId": "HSL:1240118",
    "type": "REQUEST_ROUTE_STOPS",
  },
  Object {
    "error": [SyntaxError: Unexpected string in JSON at position 56],
    "sentStoprequest": false,
    "type": "REQUEST_ROUTE_STOPS_ERROR",
  },
]
`;
